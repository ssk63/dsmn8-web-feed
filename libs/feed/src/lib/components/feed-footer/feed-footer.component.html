<div class="feed-footer">
  <p class="views-count" *ngIf="metrics?.totalReach">
    Audience size: {{ metrics?.totalReach }}
  </p>
  <div class="feed-footer__details">
    <p class="feed-footer__details-likes" *ngIf="metrics?.likes">
      {{ metrics?.likes }} likes
    </p>
    <p class="feed-footer__details-reshares" *ngIf="metrics?.shares">
      {{ metrics?.shares }} reshares
    </p>
  </div>
</div>
