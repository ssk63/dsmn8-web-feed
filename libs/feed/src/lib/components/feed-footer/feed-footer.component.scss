.feed-footer {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
  color: rgba(73, 73, 73, 0.5);
  &__details {
    @extend .feed-footer;
    &-likes {
      padding-right: 15px;
    }
  }
}
