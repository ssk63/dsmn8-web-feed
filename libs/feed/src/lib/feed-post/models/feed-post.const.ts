export const likeBtnLabel = 'Like on LinkedIn (5 pts)';
export const reshareBtnLabel = 'Reshare on LinkedIn (5 pts)';
