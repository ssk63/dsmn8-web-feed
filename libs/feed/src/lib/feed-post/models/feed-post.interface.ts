/**
 * Interface for the 'FeedPost' data
 */
export interface FeedPostEntity {
  id: string | number;
  name: string;
  content: string;
  text: string;
  boostPostMetrics: IMetrics;
  imgUrl: string;
}

export interface IMetrics {
  likes: number;
  shares: number;
  totalReach: number;
}
