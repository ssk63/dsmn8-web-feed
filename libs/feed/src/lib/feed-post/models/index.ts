export * from './feed-post.interface';
export * from './feed-post.mock-data';
