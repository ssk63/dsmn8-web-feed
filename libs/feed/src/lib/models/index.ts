export * from './feed.interface';
export * from './feed.mock.data';
export * from './feed.const';
