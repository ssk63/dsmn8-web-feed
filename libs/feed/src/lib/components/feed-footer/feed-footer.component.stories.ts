import { moduleMetadata, Story, Meta } from '@storybook/angular';
import { mockFeed } from '../../models';
import { FeedFooterComponent } from './feed-footer.component';

export default {
  title: 'FeedFooterComponent',
  component: FeedFooterComponent,
  decorators: [
    moduleMetadata({
      imports: [],
    }),
  ],
  argTypes: {
    metrics: {
      name: 'metrics',
      description: 'Sets feed social details metrics.',
      control: {
        type: 'object',
      },
      table: {
        category: 'Inputs',
      },
    },
  },
} as Meta<FeedFooterComponent>;

const Template: Story<FeedFooterComponent> = (args: FeedFooterComponent) => ({
  props: args,
});

export const Metrics = Template.bind({});
Metrics.args = {
  metrics: mockFeed[0].boostPostMetrics,
};
