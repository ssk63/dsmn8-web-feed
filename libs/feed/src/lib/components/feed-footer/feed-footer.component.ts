import { Component, Input, ViewEncapsulation } from '@angular/core';
import { IMetrics } from '../../models';

@Component({
  selector: 'dsmn8-feed-footer',
  templateUrl: './feed-footer.component.html',
  styleUrls: ['./feed-footer.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class FeedFooterComponent {
  @Input() metrics: IMetrics | undefined;
}
