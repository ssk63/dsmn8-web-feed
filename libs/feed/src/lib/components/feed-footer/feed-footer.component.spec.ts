import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { mockFeed } from '../../models';
import { FeedFooterComponent } from './feed-footer.component';

describe('FeedFooterComponent', () => {
  let component: FeedFooterComponent;
  let fixture: ComponentFixture<FeedFooterComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FeedFooterComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(FeedFooterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should render feed-post social metrics when the metrics is exists', () => {
    // arrange
    component.metrics = mockFeed[0].boostPostMetrics;
    // act
    fixture.detectChanges();
    const feedFooter = fixture.debugElement.query(
      By.css('.feed-footer')
    ).nativeElement;
    // assert
    expect(feedFooter).toMatchSnapshot();
  });
});
