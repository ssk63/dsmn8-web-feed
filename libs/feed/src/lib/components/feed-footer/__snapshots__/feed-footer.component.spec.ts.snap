// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedFooterComponent should render feed-post social metrics when the metrics is exists 1`] = `
<div
  class="feed-footer"
>
  <p
    class="views-count"
  >
     Audience size: 5307 
  </p>
  
  <div
    class="feed-footer__details"
  >
    <p
      class="feed-footer__details-likes"
    >
       2 likes 
    </p>
    
    
  </div>
</div>
`;
